feat(tutorials): tutorial for k8s video 3 (#833)
scaleway/docs-content: tutorials/get-started-kubernetes-loadbalancer/index.mdx

The next and final video in this series will look at the topic of persistent storage for a Kubernetes cluster.

<Message type="requirement">

  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [generated your API Key](/console/my-project/how-to/generate-api-key/)
  - You have followed the [first](/tutorials/get-started-containers-docker/) and [second](/tutorials/get-started-deploy-kapsule/) in this series, and have a Kubernetes Kapsule cluster running with the `whoami` application deployed on it.

</Message>

## Where were we?

In our previous tutorials, we created a [Kubernetes Kapsule](https://www.scaleway.com/en/kubernetes-kapsule/) and deployed the containerized [whoami](https://github.com/jwilder/whoami) application to it. This is what our cluster looked like at the end of the last tutorial:

<Lightbox src="scaleway-cluster.webp" alt="" />

## Why do we need a Load Balancer?

> By default, Kubernetes clusters are not exposed to the Internet. These means that external users cannot access the application deployed in our cluster.

Reviewer: ```suggestion
By default, Kubernetes clusters are not exposed to the Internet. This means that external users cannot access the application deployed in our cluster.
```

> In the previous tutorial, we solved this problem by creating a **NodePort** Service for our cluster. NodePort is a Kubernetes service which opens a port on each node in a cluster that contains pods for the specified deployment. It then forwards any external traffic received on that port to the right pods. This allowed us to go to our node’s external IP address in a browser, specifying the open port, and be served  our web application.

Reviewer 2: I think something's missing after "be served"

Author: thanks, fixed

> In the previous tutorial, we solved this problem by creating a **NodePort** Service for our cluster. NodePort is a Kubernetes service which opens a port on each node in a cluster that contains pods for the specified deployment. It then forwards any external traffic received on that port to the right pods. This allowed us to go to our node’s external IP address in a browser, specifying the open port, and be served  our web application.

Reviewer: the sentence I changed was a bit long, but maybe my suggestion doesnt make sense
```suggestion
In the previous tutorial, we solved this problem by creating a **NodePort** Service for our cluster. NodePort is a Kubernetes service which opens a port on each node in a cluster. This cluster contains pods for the specified deployment. It then forwards any external traffic received on that port to the right pods. This allowed us to go to our node’s external IP address in a browser, specifying the open port, and be served  our web application.
```

Author: This suggestion changes the meaning, but I've tried to improve the sentence a different way.

First we create the LoadBalancer Service on our cluster by connecting to it via `kubectl` and creating a YAML manifest to specify the Service.

Then, the cluster’s Cloud Controller Manager (a component of the Kapsule’s Control Plane, managed by Scaleway) takes care of creating the external Scaleway Load Balancer, and is responsible for all of its configuration and management. We can check the console though, to see that the Load Balancer has indeed been created.

We use `kubectl` to check the IP address of our Load Balancer, and then we are ready to test!

<Message type="note">

It is not possible to create the external Scaleway Load Balancer yourself via the console/API and then connect it to your cluster afterwards. You must create the LoadBalancer Service in the cluster first, so that the Scaleway Cloud Controller Manager in your Kapsule’s Control Plane can handle the creation of the Scaleway Load Balancer, with the right configuration to direct its traffic to the correct pods of your cluster.

</Message>

## Step 1: Create Load Balancer from YAML manifest

1. Open a terminal, and run the following command to check the state of your cluster:

  ```
  kubectl get all
  ```

>   You should see two pods, the NodePport Service created during the previous tutorial (along with the basic default ClusterIP Service), the deployment and the two replicasets.

Reviewer 2: ```suggestion
  You should see two pods, the NodePport Service created during the previous tutorial (along with the basic default ClusterIP Service), the deployment and the two replica sets.
```
